source/h2_stream.c

```diff
@@ -734,6 +745,13 @@ int aws_h2_stream_on_activated(struct aws_h2_stream *stream, enum aws_h2_stream_
         }
     }
     aws_h2_connection_enqueue_outgoing_frame(connection, headers_frame);
+    if (stream->base.on_metrics && !with_data) {
+        /* The frame has not sent by the connection, but the stream is considered to end sending more data. */
+        AWS_ASSERT(stream->base.metrics.send_end_timestamp_ns == 0);
+        aws_high_res_clock_get_ticks(&stream->base.metrics.send_end_timestamp_ns);
+        stream->base.metrics.sending_duration_ns =
+            stream->base.metrics.send_end_timestamp_ns - stream->base.metrics.send_start_timestamp_ns;
+    }
```

Comment:
The `send_end_timestamp` represents "the time stamp when the request has finished being sent to the HTTP level". I expected this timestamp to include everything, such as the end frame, and for the stream to not send anything after this point. 

If we assume that send_end_timestamp only applies to the data, it can be confusing when there is no data to send. This could result in a misleading metric with a very short duration, indicating that the data was sent quickly even though there was nothing to send.

Reply:
I changed the document, it's meant to represent the time stamp when the stream finish sending to the connection.

Reply:
I am still confused. How can we say the stream finished sending the data here, 'If the stream has no body, then the HEADERS frame marks the end of outgoing data'? Should we mark the end after the header is sent?